### PR TITLE
Adopt using system property based ssl configuration

### PR DESCRIPTION
**Pull request by Hakky54**

### CODE REVIEW

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java [hunk 1]
```diff
@@ -396,6 +412,31 @@ public Builder withTrustMaterial(Set<X509Certificate> certificates, TrustAnchorT
...
+            return withPlaceHolderMaterial(
+                    "javax.net.ssl.keyStore",
+                    "javax.net.ssl.keyStorePassword",
+                    "javax.net.ssl.keyStoreType",
+                    this::withIdentityMaterial
+            );
+        }
+
+        private Builder withPlaceHolderMaterial(String keyStorePathSystemProperty, String keyStorePasswordProperty, String keyStoreTypeProperty, TriConsumer<Path, char[], String> keyStorePropertyConsumer) {
+            Path keystore = Optional.ofNullable(System.getProperty(keyStorePathSystemProperty))
```
Comment (rymsha) [inline sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java:425]: `keyStore = ""` (empty string) should be treated the same as there is no system property provided.

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java [hunk 2]
```diff
@@ -396,6 +412,31 @@ public Builder withTrustMaterial(Set<X509Certificate> certificates, TrustAnchorT
...
+                    .map(Paths::get)
+                    .orElse(null);
+
+            char[] keystorePassword = Optional.ofNullable(System.getProperty(keyStorePasswordProperty))
+                    .map(String::toCharArray)
+                    .orElse(null);
+
+            String keystoreType = Optional.ofNullable(System.getProperty(keyStoreTypeProperty))
+                    .orElseGet(KeyStore::getDefaultType);
+
```
Comment (rymsha) [inline sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java:435]: for completeness `javax.net.ssl.keyStoreProvider` should be also read/used here

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java (hunk 1 above)
Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java]: Good suggestion, fixed it

File: sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java (hunk 2 above)
Comment (Hakky54) [reply sslcontext-kickstart/src/main/java/nl/altindag/ssl/SSLFactory.java]: Ah, very sharp. I added it! Thank you

**Review by rymsha — COMMENTED** [review]
LGTM

**Pull request merged**